examples/basecoin/app/app.go

```diff
@@ -58,15 +57,14 @@ func NewBasecoinApp(logger log.Logger, db dbm.DB) *BasecoinApp {
 
 	// add handlers
 	coinKeeper := bank.NewCoinKeeper(app.accountMapper)
-	coolMapper := cool.NewMapper(app.capKeyMainStore)
+	coolKeeper := cool.NewKeeper(app.capKeyMainStore, coinKeeper)
 	ibcMapper := ibc.NewIBCMapper(app.cdc, app.capKeyIBCStore)
-	stakingMapper := staking.NewMapper(app.capKeyStakingStore)
+	stakeKeeper := staking.NewKeeper(app.capKeyStakingStore, coinKeeper)
 	app.Router().
-		AddRoute("bank", bank.NewHandler(coinKeeper)).
-		AddRoute("cool", cool.NewHandler(coinKeeper, coolMapper)).
-		AddRoute("sketchy", sketchy.NewHandler()).
+		AddRoute("bank", coinKeeper.Handler).
+		AddRoute("cool", coolKeeper.Handler).
 		AddRoute("ibc", ibc.NewHandler(ibcMapper, coinKeeper)).
-		AddRoute("staking", staking.NewHandler(stakingMapper, coinKeeper))
+		AddRoute("staking", stakeKeeper.Handler)
```

Comment:
@sunnya97 as per a discussion I had with @jaekwon - we oughta' draw the line of functionality belonging to the keeper before handler . aka the main handler as well as the message handler ought to be stand-alone and take the keeper as an input. - Other complex logic on the state can still be a part of keeper however